src/domain/model/authentication/AssignmentActions.ts

```diff
@@ -0,0 +1,93 @@
+import { AssignmentAction, AssignmentId } from './AssignmentAction';
+import { AssignmentActionType } from './AssignmentActionType';
+import { UserId } from '../user/UserId';
+
+export class AssignmentActions<T, R> {
+  readonly newAssignmentActions: Array<AssignmentAction<T, R>> = [];
+
+  constructor(
+    private assignmentActions: AssignmentAction<T, R>[],
+    private getKey: (resource: R) => string | number
+  ) {}
+
+  activeAssignments(): Array<R> {
+    const groupedAssignmentActions = this.groupAssignmentActionsByResourceId();
+    return this.getActiveAssignedResources(groupedAssignmentActions);
+  }
+
+  addAssignment(resource: R, assignTo: T, actor: UserId) {
+    this.newAssignmentAction(
+      new AssignmentAction(
+        new AssignmentId(),
+        assignTo,
+        resource,
+        AssignmentActionType.ADD,
+        actor,
+        this.assignmentActions.length + 1
```

Comment:
Why do we need this?

Reply:
This is to guarantee order when actions are added sequentially (i.e an ADD followed by a REMOVE). If you solely rely on the creationTime there is a high chance they will all have the same millisecond and you won't know which one goes first. Using the order makes sure that order is always guaranteed. (In java this would be an AtomicInt, but node has a single thread so we're good)